legions/commands/more/teatime.py

```diff
@@ -0,0 +1,309 @@
+
+    # Used when checking the node's sync state.
+    block_threshold = 10
+
+    # pluginInstantiator is a dict mapping plugin names to a function
+    # instantiating and returning them.
+    pluginInstantiator = {
+        "eth1/AccountCreation": lambda arg=password: AccountCreation(arg),
+        "eth1/TxPoolContent": lambda: TxPoolContent(),
+        "eth1/HashrateStatus": lambda arg=hash_rate: HashrateStatus(arg),
+        "eth1/NetworkListening": lambda: NetworkListening(),
+        "eth1/NodeSync": lambda arg1=infura_url, arg2=block_threshold: NodeSync(
+            arg1, arg2
+        ),
+        "eth1/SHA3Consistency": lambda arg1=test_input, arg2=test_output: SHA3Consistency(
+            arg1, arg2
+        ),
+        "eth1/OpenAccounts": lambda arg=infura_url: OpenAccounts(arg),
+        "eth1/AccountUnlock": lambda arg1=infura_url, arg2=word_list, arg3=skip_below: AccountUnlock(
+            arg1, arg2, arg3
+        ),
+        "eth1/NodeVersion": lambda: NodeVersion(),
+        "eth1/PeerlistLeak": lambda: PeerlistLeak(),
+        "eth1/MiningStatus": lambda arg=should_mine: MiningStatus(arg),
+        "eth1/PeerCountStatus": lambda arg=minimum_peercount: PeerCountStatus(arg),
+        "eth1/PeerlistManipulation": lambda arg=test_enode: PeerlistManipulation(arg),
+        # Geth
+        "eth1/GethNodeInfo": lambda: GethNodeInfo(),
+        "eth1/GethAccountImport": lambda arg=password: GethAccountImport(arg),
+        "eth1/GethStartWebsocket": lambda: GethStartWebsocket(),
+        "eth1/GethStopWebsocket": lambda: GethStopWebsocket(),
+        "eth1/GethTxPoolInspection": lambda: GethTxPoolInspection(),
+        "eth1/GethTxPoolStatus": lambda: GethTxPoolStatus(),
+        "eth1/GethStartRPC": lambda: GethStartRPC(),
+        "eth1/GethStopRPC": lambda: GethStopRPC(),
+        "eth1/GethDatadir": lambda: GethDatadir(),
+        # Parity
+        "eth1/ParityGasCeiling": lambda arg=gasceiling: ParityGasCeiling(arg),
+        "eth1/ParityDevLogs": lambda: ParityDevLogs(),
+        "eth1/ParityGasFloor": lambda arg=gasfloor: ParityGasFloor(arg),
+        "eth1/ParityUpgrade": lambda: ParityUpgrade(),
+        "eth1/ParityTxPoolStatistics": lambda: ParityTxPoolStatistics(),
+        "eth1/ParityTxCeiling": lambda arg=gas_limit: ParityTxCeiling(arg),
+        "eth1/ParityMinGasPrice": lambda arg=gas_price: ParityMinGasPrice(arg),
+        "eth1/ParitySyncMode": lambda arg=mode: ParitySyncMode(arg),
+        "eth1/ParityChangeCoinbase": lambda arg=author: ParityChangeCoinbase(arg),
+        "eth1/ParityChangeTarget": lambda arg=target_chain: ParityChangeTarget(arg),
+        "eth1/ParityChangeExtra": lambda arg=extra_data: ParityChangeExtra(arg),
+        "eth1/ParityDropPeers": lambda: ParityDropPeers(),
+    }
+
+    # Contains plugin names added with the `add` command.
+    addedPlugins = set("eth1/NodeVersion")
+
+    def __init__(self) -> None:
+        # Set node_type.
+        if "Geth" in w3.clientVersion:
```

Comment:
Making sure the node type detection is a bit more robust here because you never know what `clientVersion` will give you.
```suggestion
        if "geth" in w3.clientVersion.lower():
```